transports/transport.h

```diff
@@ -0,0 +1,206 @@
+ * it must implement the following methods and callbacks for the gateway:
+ * 
+ * - \c init(): this is called by the gateway as soon as your transport
+ * plugin is started; this is where you should setup your transport plugin
+ * (e.g., static stuff and reading the configuration file);
+ * - \c destroy(): on the other hand, this is called by the gateway when it
+ * is shutting down, and your transport plugin should too;
+ * - \c get_api_compatibility(): this method MUST return JANUS_TRANSPORT_API_VERSION;
+ * - \c get_version(): this method should return a numeric version identifier (e.g., 3);
+ * - \c get_version_string(): this method should return a verbose version identifier (e.g., "v1.0.1");
+ * - \c get_description(): this method should return a verbose description of your transport plugin (e.g., "This is my avian carrier transport plugin for the Janus API");
+ * - \c get_name(): this method should return a short display name for your transport plugin (e.g., "My Amazing Transport");
+ * - \c get_package(): this method should return a unique package identifier for your transport plugin (e.g., "janus.transport.mytransport");
+ * - \c send_message(): this method asks the transport to send a message (be it a response or an event) to a client on the specified transport;
+ * - \c session_timeout(): this method notifies the transport that a Janus session originated here timed out;
+ * 
+ * All the above methods and callbacks are mandatory: the Janus core will
+ * reject a transport plugin that doesn't implement any of the
+ * mandatory callbacks.
+ * 
+ * The gateway \c janus_transport_callbacks interface is provided to a
+ * transport plugin, together with the path to the configurations files
+ * folder, in the \c init() method. This path can be used to read and
+ * parse a configuration file for the transport plugin: the transport
+ * plugins we made available out of the box use the package name as a
+ * name for the file (e.g., \c janus.transport.http.cfg for the HTTP/HTTPS
+ * transport plugin), but you're free to use a different one, as long
+ * as it doesn't collide with existing ones. Besides, the existing transport
+ * plugins use the same INI format for configuration files the gateway
+ * uses (relying on the \c janus_config helpers for the purpose) but
+ * again, if you prefer a different format (XML, JSON, etc.) that's up to you. 
+ * 
+ * \ingroup transportapi
+ * \ref transportapi
+ */
+
+#ifndef _JANUS_TRANSPORT_H
+#define _JANUS_TRANSPORT_H
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include <unistd.h>
+#include <inttypes.h>
+
+#include <glib.h>
+#include <jansson.h>
+
+
+/*! \brief Version of the API, to match the one transport plugins were compiled against */
+#define JANUS_TRANSPORT_API_VERSION	4
+
+/*! \brief Initialization of all transport plugin properties to NULL
+ * 
+ * \note All transport plugins MUST add this as the FIRST line when initializing
```

Comment:
We have a similar approach for plugins. Someone else made me notice that this might be unneeded. Anyway, this looked like a more convenient and "prettier" way of settings things up than asking developers to memset the struct to 0.